internal/tweets/edit.go

```diff
@@ -3,39 +3,30 @@ package tweets
 import (
 	"database/sql"
 	"encoding/json"
-	"fmt"
 	"net/http"
 
 	"Twitter_like_application/internal/services"
 
-	"github.com/go-playground/validator/v10"
 	"github.com/gorilla/mux"
 )
 
-type EditTweetRequest struct {
-	Text string `json:"text" validate:"required,checkTweetText"`
-	Visibility
-}
+type editTweetRequest = createTweetRequest
 
 func (s *Service) Edit(w http.ResponseWriter, r *http.Request) {
 	tweetID := mux.Vars(r)["id_tweet"]
 	userID := r.Context().Value("userID").(int)
-	tweetValid := &TweetValid{
-		Validate: validator.New(),
-		ValidErr: make(map[string]string),
-	}
-	if err := RegisterTweetValidations(tweetValid); err != nil {
-		fmt.Println(err)
-	}
-
-	var request EditTweetRequest
+	var req editTweetRequest
 	var tweet Tweet
-	err := json.NewDecoder(r.Body).Decode(&request)
+	err := json.NewDecoder(r.Body).Decode(&req)
 	if err != nil {
 		services.ReturnErr(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	if !request.isValid() {
+	if err := req.validate(); err != nil {
+		services.ReturnErr(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	if !req.isValid() {
```

Comment:
this is redundant, you have this check inside the `.validate()` that you call 4 lines higher